packages/shell-api/src/database.ts

```diff
@@ -144,6 +149,21 @@ export default class Database extends ShellApiWithMongoClass {
     return this._cachedCollectionNames;
   }
 
+  async _getCollectionNamesWithTypes(options?: ListCollectionsOptions): Promise<CollectionNamesWithTypes[]> {
+    const collections = await this._listCollections({}, { ...options, nameOnly: true });
+    const typesToBages: any = {
+      timeseries: '[time-series]',
+      view: '[view]'
+    };
+
+    this._cachedCollectionNames = collections.map((collection: any) => collection.name);
+
+    return collections.map((collection: any) => ({
+      name: collection.name,
+      badge: typesToBages[collection.type] ? typesToBages[collection.type] : ''
```

Comment:
```suggestion
      badge: typesToBages[collection.type] ?? ''
```

(I think that’s just a bit more common in the mongosh source, at least by now)